tests/models/test_taskmixin.py

```diff
@@ -252,7 +256,10 @@ def test_set_setup_teardown_ctx_dependencies_using_decorated_tasks(dag_maker):
         t3 = make_task("t3", type_="decorated")
         setuptask = make_task("setuptask", type_="decorated", setup_=True)
         teardowntask = make_task("teardowntask", type_="decorated", teardown_=True)
-        with setuptask >> teardowntask:
+        with setuptask >> teardowntask as scope:
+            scope.add_task(t1)
+            scope.add_task(t2)
+            scope.add_task(t3)
```

Comment:
I think we should move instantiating the tasks into the context manager instead of using `add_task`

Reply:
see https://github.com/apache/airflow/pull/33102/files#r1284123448